PythonCodeConventions.md

```diff
@@ -1,25 +1,33 @@
 Python Code Conventions
 =======
 
-PEP8 is the code style guide for Python code used by the main python distribution.
-This style guide can be found at:
-https://www.python.org/dev/peps/pep-0008/
+General
+---
+Ultimaker adheres to the PEP8 coding guidelines as described here: https://www.python.org/dev/peps/pep-0008/,
 
-Our coding conventions follow the PEP style guide, with the exceptions stated in the generic code_conventions.md provided in this repository. This also means that for rules regarding class naming, documentation, etc., the rules described in the general code convention document MUST be used.
+** Important: **
+Also check the code_conventions.md file for the generic rules, regardless of language.
 
-For convenience, these are re-listed here. In case of conflict, the generic code_conventions are considered to be leading.
-* Always use spaces. While PEP8 allows tabs for backwards compatibility, we do not.
-* Strings are double quotes. While Python allows single and double quoted strings, and PEP8 only says "pick a rule and stick to it". We decided to do double quotes to match C++.
-* "Maximum line length" - PEP8 specifies a maximum line length of 79 characters. At Ultimaker we have no official line length limit, although we strive to keep lines below 120 characters.
-* Naming convention:
-  * Function names in PEP8 are in lower_case_with_underscores. At Ultimaker we decided to match our C++ coding convention and use mixedCase (CapitalizedWords by initial lowercase character).
-  * Parameters, variables and members follow the lower_case_with_underscores convention. PEP8 has no recommendation on this.
+With the following adjustments, exceptions and specific choices:
+* See the 'line breaking' section below for specific suggested PEP8 style choices where function calls and definitions exceed the soft line length limit of 120 characters.
+* MUST apply the following string formatting:
+  * Multiple elements:
+``` python
+"{foo}_{bar}".format(foo = "foo", bar = "bar")
```

Comment:
I have to say that I've always preferred to have keyword arguments written with no extra spaces. I like it because it visually groups each argument as a compact unit and the spaces between args neatly separate them.

Reply:
@ckielstra @ChrisTerBeke @sedwards2009 Although I tend to agree with you on not adding the extra spaces, they were added to support increased readability for dyslexic people of which Ultimaker employs quite a few. This was mentioned in earlier comments, but probably drowned with a lot of other comments :)

Reply:
And the problem with the spaces is that pep8 is unclear on what should be done. I personally think without spaces is more readable for my dyslectic mind. (as spaces separate the arguments).

Reply:
@daid: formatting and logging; yes, this was not intended for use with logging, for the reasons you provided. Will add that to the description.
Concerning spaces: argl, so one says spaces help and the other says no spaces help... I'll leave it up to you (@daid @nallath) to figure something out :)

Reply:
I don't really care. In cloud we use spaces because that's what happens in Cura (so I took along the code style). We run flake8 for checking the code formatting in an automated way, so we can simply change the rules if something is decided here.

Reply:
@ChrisTerBeke Let's stick to PEP8 then.